src/main/js/config/ConfigPage.tsx

```diff
@@ -41,34 +41,46 @@ const DisplayOption: FC<DisplayOptionProps> = ({ value, toggle, label }) => {
 
 const ConfigPage: FC = () => {
   const [t] = useTranslation("plugins");
-  const { isDisabled, toggleDisabled } = useDisabledCategories();
+  const { isDisabled, setCategories } = useDisabledCategories();
   const binder = useBinder();
+  const [changedCategories, setChangedCategories] = useState<{ [key: string]: boolean }>({});
   const extensions: ExtensionProps[] = binder.getExtensions("landingpage.mydata");
 
+  const changeExtension = (category: string) => (enabled: boolean) => {
+    const newValue = { ...changedCategories, [category]: enabled };
+    setChangedCategories(newValue);
+  };
+
+  const shouldBeEnabled = (category: string) => {
+    return Object.keys(changedCategories).includes(category) ? changedCategories[category] : !isDisabled(category);
+  };
+
+  const submitChanges = () => {
+    setCategories(changedCategories);
+    setChangedCategories({});
+  };
+
   return (
     <Page title={t("scm-landingpage-plugin.config.title")} subtitle={t("scm-landingpage-plugin.config.subtitle")}>
-      <DisplayOption
-        label={t("scm-landingpage-plugin.favoriteRepository.title")}
-        value={isDisabled("favoriteRepository")}
-        toggle={() => toggleDisabled("favoriteRepository")}
-      />
-      <DisplayOption
-        label={t("scm-landingpage-plugin.mytasks.title")}
-        value={isDisabled("mytasks")}
-        toggle={() => toggleDisabled("mytasks")}
-      />
-      <DisplayOption
-        label={t("scm-landingpage-plugin.myevents.title")}
-        value={isDisabled("myevents")}
-        toggle={() => toggleDisabled("myevents")}
-      />
+      {["favoriteRepository", "mytasks", "myevents"].map(category => (
+        <DisplayOption
+          label={t(`scm-landingpage-plugin.${category}.title`)}
+          value={shouldBeEnabled(category)}
+          toggle={changeExtension(category)}
+        />
```

Comment:
The component is missing a [key](https://reactjs.org/docs/lists-and-keys.html#keys).

```suggestion
        <DisplayOption
          key={category}
          label={t(`scm-landingpage-plugin.${category}.title`)}
          value={shouldBeEnabled(category)}
          toggle={changeExtension(category)}
        />
```